napari/layers/points/points.py

```diff
@@ -964,9 +964,12 @@ def on_key_press(self, event):
                     self._paste_data()
             elif event.key == 'a':
                 if self._mode == Mode.SELECT:
-                    self.selected_data = self._indices_view[
-                        list(range(len(self._data_view)))
-                    ]
+                    if len(self._data_view) > 0:
+                        self.selected_data = self._indices_view[
+                            list(range(len(self._data_view)))
```

Comment:
why isn't this just `self._indices_view[:len(self._data_view)]`?

then you wouldn't even need an if-statement:
```python
In [1]: import numpy as np

In [2]: l = np.array(list(range(10, 20)))

In [3]: l[list(range(5))]
Out[3]: array([10, 11, 12, 13, 14])

In [4]: l[:5]
Out[4]: array([10, 11, 12, 13, 14])

In [5]: l[:0]
Out[5]: array([], dtype=int64)
```

Reply:
fixed